mne/io/boxy/boxy.py

```diff
@@ -0,0 +1,404 @@
+        aux1 = raw_data['aux-1'].to_numpy() if 'aux-1' in raw_data.columns else []
+        digaux = raw_data['digaux'].to_numpy() if 'digaux' in raw_data.columns else []
+        bias = np.zeros((detect_num,len(raw_data)))
+
+       ###loop through detectors###
+        for i_detect in detectors[0:detect_num]:
+            
+            ###older boxy files don't seem to keep track of detector bias###
+            ###probably due to specific boxy settings actually###
+            if 'bias-A' in raw_data.columns:
+                bias[detectors.index(i_detect),:] = raw_data['bias-' + i_detect].to_numpy()
+                
+            ###loop through data types###
+            for i_data in data_types:
+                ###loop through sources###
+                for i_source in sources:                    
+                    ###where to store our data###
+                    index_loc = detectors.index(i_detect)*source_num + (i_source-1)                    
+                    ###need to treat our filetypes differently###
+                    if filetype == 'non-parsed':
+                        
+                        ###filetype saves timepoints in groups###
+                        ###this should account for that###
+                        time_points = np.arange(i_source-1,int(record[-1])*source_num,source_num)
+                        
+                        ###determine which channel to look for###
+                        channel = i_detect + '-' + i_data
+                        
+                        ###save our data based on data type###
+                        if data_types.index(i_data) == 0:
+                            raw_ac[index_loc,:] = raw_data[channel][time_points].to_numpy()
+                        elif data_types.index(i_data) == 1:
+                            raw_dc[index_loc,:] = raw_data[channel][time_points].to_numpy()
+                        elif data_types.index(i_data) == 2:
+                            raw_ph[index_loc,:] = raw_data[channel][time_points].to_numpy()
+                    elif filetype == 'parsed':                    
+                        ###determine which channel to look for###
+                        channel = i_detect + '-' + i_data + str(i_source)
+                        
+                        ###save our data based on data type###
+                        if data_types.index(i_data) == 0:
+                            raw_ac[index_loc,:] = raw_data[channel].to_numpy()
+                        elif data_types.index(i_data) == 1:
+                            raw_dc[index_loc,:] = raw_data[channel].to_numpy()
+                        elif data_types.index(i_data) == 2:
+                            raw_ph[index_loc,:] = raw_data[channel].to_numpy()
+     
+        ###now combine our data types into a single array with the data###
+        data = np.append(raw_ac, np.append(raw_dc, raw_ph, axis=0),axis=0)
```

Comment:
This is the problem -- you must enter data into the `data` object you are provided. Quoting `mne/io/base.py`:
```
        data : ndarray, shape (len(idx), stop - start + 1)
            The data array. Should be modified inplace.
```
You currently do not modify it in place. If I change this line to `data_ = ...`, and then ...